src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java

```diff
@@ -60,21 +64,36 @@ public final void publishWithDelay(Message message, long delayMilliseconds) thro
         }
     }
 
-    public final void publish(Message message) throws Exception {
-        publish(message, MessageProperties.MINIMAL_PERSISTENT_BASIC);
+    public final void publishWithExpiry(final Message message, final long expiryInMs) throws Exception {
+        val expiresAt = Instant.now().toEpochMilli() + expiryInMs;
+        val properties = new AMQP.BasicProperties.Builder()
+                .deliveryMode(2)
+                .headers(ImmutableMap.of(MESSAGE_EXPIRY_TEXT, expiresAt))
+                .build();
+        publish(message, properties);
+    }
+
+    public final void publish(final Message message) throws Exception {
+        val properties = new AMQP.BasicProperties.Builder()
```

Comment:
Keep current implementation of this intact since timestamp addtiion is happening in the called method